spec/booking_system_spec.rb

```diff
@@ -0,0 +1,328 @@
+require_relative 'spec_helper'
+
+describe 'BookingSystem class' do
+
+  let (:today) { Date.today }
+  let (:res1_checkout) { today + 3 }
+  let (:res2_checkout) { today + 7}
+  let (:today_minus_2) {today - 2}
+
+  let (:reservation1) {
+    check_in = today
+    check_out = check_in + 3
+    Hotel::Reservation.new(Hotel::DateRange.new(check_in, check_out), 1)
+  }
+
+  let (:reservation2) {
+    check_in = today + 2
+    check_out = check_in + 5
+    Hotel::Reservation.new(Hotel::DateRange.new(check_in, check_out), 2)
+  }
+
+  let (:hotel_booker) {
+    Hotel::BookingSystem.new(all_reservations: [reservation1, reservation2])
+  }
+
+  let (:captures_both_res_dates) { Hotel::DateRange.new(today, res2_checkout)}
+  let (:overlap_res_1) {Hotel::DateRange.new(today, today + 1)}
+  let (:overlap_res_2) { Hotel::DateRange.new(res1_checkout, res2_checkout) }
```

Comment:
You end up using this var `overlap_res_2` only twice in the whole file, and even though it's lazy-loading with let so it's programmatically efficient, it's still here declared at the top of the file... The one of two references to `overlap_res_2` is on line 120, so in order to find out what `overlap_res_2` was, I had to scroll a lot to figure out the definition!

Reply:
Sorry, Dee! Will definitely note for later!